### ISSUE TITLE
MA0053: Class should be sealed on partial method

### ISSUE THREAD

**Issue by royberris** [meziantou/Meziantou.Analyzer#426]
**Context**
Rule identifier: _MA0053_
Target Framework: _.NET 7.0_
Language version: _11.0_

**Describe the bug**
A partial class should be sealed according to MA0053. This does show up with `dotnet build`. This shows up on EF Core generated migration files. With signature:

```
public partial class Initial : Migration
```

Console says:
```
src\...\Persistence\Migrations\20220926125445_Initial.cs(7,22): warning MA0053: Make class sealed
```

**Additional context**
Analyzer version 1.0.756

**Comment by royberris** [comment]
Weird thing is, if I double click the warning in VS it disappears. Feels like it is rerunning the analyzer and deciding that running it from the VS17.4.2 it is no longer an issue? It might not be a problem with the analyzer after all, not sure

**Comment by meziantou** [comment]
Visual Studio is sometime flaky with analyzers. `dotnet build` is more reliable.

In the case of a partial class, MA0053 is reported on all locations. If you have an editor config that disable the rules in specific folders, maybe this can be problematic.

Could you create a small repro project, so I can have a look?

**Comment by royberris** [comment]
Hi @meziantou 

I managed to reproduce it in a small project. This is the same set-up as it is in the larger solution. Same editorconfig, same global supression rule that goes on top of the migration's namespace. Running `dotnet build` return a warning about MA0053.

Running SDK `7.0.100`

https://github.com/royberris/MeziantouAnalyzerMA53

**Comment by royberris** [comment]
> You have an editor config that disable the rules in specific folders, maybe this can be problematic.

I do not have a rule to disable MA0053 in specific folders. Only some global supressions, like shown in the example project.

**Comment by meziantou** [comment]
The `Foo` class is defined in 2 files (partial class) and is not `sealed`. The MA0053 rule is configured to report on public class: `MA0053.public_class_should_be_sealed = true`. So, the analyzer must report a diagnostic.

`20221208072609_Foo.Designer.cs` is generated (`// <auto-generated />` at the top of the file), so Roslyn doesn't report the error in this file. But it does report on the other file.

On my side the warning is visible using dotnet build and VS as expected:

<img width="379" alt="image" src="https://user-images.githubusercontent.com/509220/206617142-1250de20-6f5b-4dc3-8305-91b9b38cd661.png">

**Comment by royberris** [comment]
Right, so I should update the code manually so that it is a `public sealed partial class`? I didn't know a `sealed partial class` was a thing, but it makes sense. Learned something today, sorry for the misunderstanding.
